app/models/kudo.rb

```diff
@@ -3,6 +3,7 @@ class Kudo < ActiveRecord::Base
   belongs_to :commentable, :polymorphic => true
 
   validate :cannot_be_author
+  validate :guest_cant_restricted
```

Comment:
As long as you're editing, this method name is kind of opaque. Something like `guest_cannot_kudos_restricted_work` would probably be better -- it's longer, but it's very clear about what it's checking, and the examples in the [custom methods section of the RoR guide](http://guides.rubyonrails.org/active_record_validations.html#custom-methods) don't shy away from long names, either.